code/scripting/api/objs/control_binding.cpp

```diff
@@ -6,11 +6,11 @@
 namespace scripting {
 namespace api {
 
-cci_h::cci_h() { idx = CCFG_MAX; }
+cci_h::cci_h() { idx = IoActionId::CCFG_MAX; }
 
-cci_h::cci_h(int n_id) { idx = static_cast<IoActionId>(n_id); }
+cci_h::cci_h(int n_id) { if (n_id < 0 || n_id > static_cast<int>(IoActionId::CCFG_MAX)) idx = IoActionId::CCFG_MAX; else idx = static_cast<IoActionId>(n_id); }
 
-bool cci_h::IsValid() { return (idx > -1 && idx < IoActionId::CCFG_MAX); }
+bool cci_h::IsValid() { return idx < IoActionId::CCFG_MAX; }
```

Comment:
Nitpick.  Enumerations should not be treated as integers and instead be treated as unique identifiers or states, this means that the >, >=, <, and <= operators should not be used on enumerations.

This is often overlooked for various reasons, and we're not that uptight about this rule, either. As an example we exploit that enums *can* be treated as int's to check for sub-ranges within the enum (specifically the joystick axes).

I'll Allow It.

Reply:
I mean, it's never too late to start doing things the right way.  Shall I do `idx != IoActionId::CCFG_MAX;` instead?